client/src/ui/FocusDirector.lua

```diff
@@ -1,23 +1,45 @@
--- use in tandem with Focusable.lua
+-- FocusDirector Mixin
+--
+-- Manages keyboard/controller focus for a container of focusable objects.
+-- Use in tandem with Focusable.lua
+--
+-- Usage:
+-- 1. Apply FocusDirector(container) to enable focus management
+-- 2. Call container:setFocus(focusableObject, callback) to focus an object
+-- 3. The focused object receives a dynamically assigned yieldFocus() method
+-- 4. When yieldFocus() is called, focus is cleared and callback is executed
+--
+-- The director assigns yieldFocus() to the focused object, which:
+-- - Sets object.hasFocus = false
+-- - Clears director.focused = nil
+-- - Executes optional callback function
 
-local function directsFocus(uiElement)
-    uiElement.focused = nil
-    uiElement.setFocus = function(table, focusable, callback)
-      if table.focused then
-        table.focused.hasFocus = false
-      end
-      table.focused = focusable
-      if focusable then
-        focusable.hasFocus = true
-        focusable.yieldFocus = function()
-          focusable.hasFocus = false
-          table.focused = nil
-          if callback then
-            callback()
-          end
+local function directsFocus(object)
```

Comment:
I'm kind of disagreeing with the rename to `object` here, I would hope none ever tries to use this for generic objects.

Reply:
It was called uiElement before I think, that seemed good for me.
And yes, in my resizable UI branch I did integrate that in some way. We'll get to that some day.